### PR TITLE
fix(injectors): target the seen IP in automatic mode (#417)

### PR DESCRIPTION
**Pull request by Seb-MIGUEL**
Fixes #417 (reported as OpenAEV-Platform/filigran-private#257).

An endpoint was displayed at one address and scanned at another: the endpoint screen shows `winterfell` as `74.234.220.121` (its **seen IP**) while the nmap output showed the scan hitting `192.168.56.11`.

`automatic` is the default targeted-asset property across the injectors, and it never read `asset_seen_ip` - it fell straight through to the first entry of `asset_ips`, a local address. The field was available on the payload all along (the explicit `seen_ip` selector already reads it), so the platform was reporting one address and attacking another with nothing in the UI to reveal the difference.

### Change

In `extract_property_target_value`, try the seen IP before the local IPs:

1. agentless + hostname -> hostname *(unchanged)*
2. **seen IP, when valid** *(new)*
3. first valid local IP *(unchanged, now the fallback)*

`is_valid_ip` still filters loopback / unspecified / link-local, so an endpoint whose seen IP is unusable falls back exactly as before. The explicit `hostname` / `seen_ip` / `local_ip` selectors keep their precise meaning - only `automatic` changes.

### Follow-up hardening

`asset_seen_ip` is optional in payloads, so `extract_property_target_value` can call `is_valid_ip(None)` for assets without a seen IP. `is_valid_ip` now also catches `TypeError` (in addition to the `ValueError` that `ipaddress.ip_address` already raises for `None` / empty strings) and its type hint is widened to `Optional[str]`, making the optional-input contract explicit and defensive across input types. This addresses the Copilot review note on `is_valid_ip`.

### Verification

Tests added, run against the code **with and without** the original fix:

| test | without the fix |
|---|---|
| prefers the seen IP over the local IP | `AssertionError: '192.168.56.11' != '74.234.220.121'` - the reported IPs exactly |
| falls back when the seen IP is loopback | passes (guards the fallback against regression) |

Plus regression tests for the follow-up hardening: `automatic` falls back to the local IP when `asset_seen_ip` is `None`, and `is_valid_ip` returns `False` for `None` / empty / invalid values.

`18 passed` on the whole `injector_common` suite. `black`, `isort --profile black`, and `flake8` clean on the touched files.

### CODE REVIEW

File: injector_common/injector_common/targets.py [hunk 1]
```diff
@@ -230,7 +230,12 @@ def extract_property_target_value(asset: Dict) -> Optional[Tuple[str, str]]:
         """
         Extract target value from asset based on conditions:
         - Agentless + hostname => hostname
-        - Otherwise => first valid IP
+        - Otherwise => seen IP, else first valid local IP
```
[the remark this answers is not in the document]
Comment (SamuelHassine) [reply injector_common/injector_common/targets.py:239]: Good catch on the robustness of `is_valid_ip` with optional payload fields. One clarification on the exact exception: on CPython, `ipaddress.ip_address(None)` raises `ValueError` (not `TypeError`), because `ip_address` stringifies its argument (`str(None)` -> `"None"`) and the parse fails with `AddressValueError`, a `ValueError` subclass. So the existing `except ValueError` already treated a missing/None `asset_seen_ip` as invalid - this is exercised today by the passing tests whose assets carry no `asset_seen_ip` key (`asset_local_ip`, `empty_asset_ips`), which reach the IP branch via `automatic` and fall through correctly.

That said, your underlying concern is valid, so I hardened the helper anyway: `is_valid_ip` now also catches `TypeError` (defensive against any non-stringifiable / unexpected type) and its type hint is widened to `Optional[str]` to make the optional-input contract explicit. I also added regression tests: `automatic` targeting falls back to the local IP when `asset_seen_ip` is `None`, and `is_valid_ip` returns `False` for `None` / empty / invalid values. Done in 27f2db1.

**Review by SamuelHassine — APPROVED** [review]
Approving after an independent senior review of the full changed files.

The core fix is correct: `automatic` targeting now tries the valid seen IP before the local IPs, matching what the endpoint screen displays, while `is_valid_ip` still filters loopback / unspecified / link-local so an unusable seen IP falls back exactly as before. The explicit `hostname` / `seen_ip` / `local_ip` selectors are untouched, and `extract_property_target_value` is only consumed via the `automatic` path in `get_target`, whose calls are wrapped in a try/except in `process_targets`, so no injector regresses.

I added a small hardening commit and its regression tests (see the follow-up section in the description and the reply on the `is_valid_ip` thread). Local gates pass: `18 passed`, `black` / `isort --profile black` / `flake8` clean. CI is fully green.
